### DISCUSSION TITLE
types.Date is giving error.

### DISCUSSION THREAD

**Discussion by abhiram2600** [mobxjs/mobx-state-tree#1696] [Q&A] [answered]
I have a model which has props:- 

`comments: types.array(
      types.model({
        commentBy: types.string,
        comment: types.string,
        commentTimeStamp: types.Date,
      })
    ),`

when i fill in values then i get this error:-
`Error: [mobx-state-tree] Error while converting [{"comment":"abcs","commentTimeStamp":{"seconds":1618557929,"nanoseconds":191000000},"commentBy":"abhi"}] to AnonymousModel[]:
  at path “/0/commentTimeStamp” snapshot {"seconds":1618557929,"nanoseconds":191000000} is not assignable to type: Date (Value is not a Date or a unix milliseconds timestamp).`

**Comment by EmilTholin** [comment, marked as answer]
Hi @abhiram2600!

It looks like the value you are trying to pass to `commentTimeStamp` is an object with properties `seconds` and `nanoseconds`. You could e.g. add a custom `preProcessSnapshot` to just take the `seconds` property and multiply it with 1000 to get milliseconds, and MST will be happy.

```js
const MyModel = types
  .model("MyModel", {
    comments: types.array(
      types.model({
        commentBy: types.string,
        comment: types.string,
        commentTimeStamp: types.Date
      })
    )
  })
  .preProcessSnapshot((snapshot) => ({
    ...snapshot,
    comments: snapshot.comments.map((comment) => ({
      ...comment,
      commentTimeStamp: comment.commentTimeStamp.seconds * 1000
    }))
  }));

const myModel = MyModel.create({
  comments: [
    {
      comment: "abcs",
      commentTimeStamp: { seconds: 1618557929, nanoseconds: 191000000 },
      commentBy: "abhi"
    }
  ]
});

console.log(myModel.comments[0].commentTimeStamp);
// Fri Apr 16 2021 09:25:29 GMT+0200 (Central European Summer Time)
```

```
